pkg/scalers/redis_scaler.go

```diff
@@ -46,31 +47,36 @@ type redisScaler struct {
 }
 
 type redisConnectionInfo struct {
-	addresses        []string
-	username         string
-	password         string
-	sentinelUsername string
-	sentinelPassword string
-	sentinelMaster   string
-	hosts            []string
-	ports            []string
-	enableTLS        bool
-	unsafeSsl        bool
-	cert             string
-	key              string
-	keyPassword      string
-	ca               string
+	Addresses        []string `keda:"name=address;addresses,       order=triggerMetadata;authParams;resolvedEnv"`
+	Username         string   `keda:"name=username,       order=triggerMetadata;resolvedEnv;authParams"`
+	Password         string   `keda:"name=password,       order=triggerMetadata;resolvedEnv;authParams"`
+	SentinelUsername string   `keda:"name=sentinelUsername,       order=triggerMetadata;authParams;resolvedEnv"`
+	SentinelPassword string   `keda:"name=sentinelPassword,       order=triggerMetadata;authParams;resolvedEnv"`
+	SentinelMaster   string   `keda:"name=sentinelMaster,       order=triggerMetadata;authParams;resolvedEnv"`
+	Hosts            []string `keda:"name=host;hosts,       order=triggerMetadata;resolvedEnv;authParams"`
+	Ports            []string `keda:"name=port;ports,       order=triggerMetadata;resolvedEnv;authParams"`
+	EnableTLS        bool     `keda:"name=enableTLS;tls,       order=triggerMetadata;authParams, optional, default=false"`
```

Comment:
Could you help to elaborate more on why are we supporting an array of name here ?

Reply:
Because there are different names for different scale types, they will be converted to the same property in the end. For example, the 'host' parameter of RedisList and the 'hosts' parameter of RedisListCluster are both converted to the 'hosts' property of RedisConnectionInfo nowadays. If we don't support an array of names, we will need to create two different RedisConnectionInfo structs or do duplicated work in conversion.

Reply:
I like it, given the original parsing code for redis scaler supported that and it brings no breaking change to other scaleds, it seems like a good idea to me

Reply:
Thanks for the clear explanation @SpiritZhou. I don't have any problem with the implementation, just concerned about the use case. But it seems we have a concrete use for it, I'm all good